source/extensions/http/cache/file_system_http_cache/stats.h

```diff
@@ -19,19 +19,44 @@ namespace FileSystemHttpCache {
  *
  * Drift will eventually be reconciled at the next pre-cache-purge measurement.
  **/
-#define ALL_CACHE_STATS(COUNTER, GAUGE)                                                            \
+
+#define ALL_CACHE_STATS(COUNTER, GAUGE, HISTOGRAM, TEXT_READOUT, STATNAME)                         \
   COUNTER(eviction_runs)                                                                           \
   GAUGE(size_bytes, NeverImport)                                                                   \
   GAUGE(size_count, NeverImport)                                                                   \
   GAUGE(size_limit_bytes, NeverImport)                                                             \
-  GAUGE(size_limit_count, NeverImport)
+  GAUGE(size_limit_count, NeverImport)                                                             \
+  STATNAME(cache)                                                                                  \
+  STATNAME(cache_path)
 // TODO(ravenblack): Add other stats from DESIGN.md
 
+#define COUNTER_HELPER_(NAME)                                                                      \
+  , NAME##_(                                                                                       \
+        Envoy::Stats::Utility::counterFromStatNames(scope, {prefix_, stat_names.NAME##_}, tags_))
+#define GAUGE_HELPER_(NAME, MODE)                                                                  \
+  , NAME##_(Envoy::Stats::Utility::gaugeFromStatNames(                                             \
+        scope, {prefix_, stat_names.NAME##_}, Envoy::Stats::Gauge::ImportMode::MODE, tags_))
+#define STATNAME_HELPER_(NAME)
+
+MAKE_STAT_NAMES_STRUCT(CacheStatNames, ALL_CACHE_STATS);
+
 struct CacheStats {
-  ALL_CACHE_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT)
+  CacheStats(const CacheStatNames& stat_names, Envoy::Stats::Scope& scope,
+             Stats::StatName&& cache_path)
```

Comment:
rvalue-ref not necessary for StatName -- it's just a single pointer. Suggest just passing by value.

Think of StatName like a string_view -- its storage needs to be owned somewhere else, but it's cheap to pass them around.

Reply:
Done. Thanks for the analogy, that makes it clearer.